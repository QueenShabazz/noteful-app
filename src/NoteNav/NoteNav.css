.NoteDetails{
    display: flex;
    flex-direction: row;
}

.details{
    margin: 0 50px;
}