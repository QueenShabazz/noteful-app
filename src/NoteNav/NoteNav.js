import React from 'react'
import ApiContext from '../ApiContext'
import {findNote, findFolder} from '../notes-helper'
import './NoteNav.css'
import PropTypes from 'prop-types'

export default class NoteNav extends React.Component{
    static defaultProps={
        history: {
            goBack: ()=> {}
        },
        match:{
            params:{}
        }
    }
    static contextType = ApiContext;
    render() {
        const {notes, folders,} = this.context
        const {noteId} = this.props.match.params
        const note = findNote(notes, noteId) || {}
        const folder = findFolder(folders, note.folderId)
        return(
            <div className="NoteDetails">
                <div>
                    <button type="button"
                        onClick={()=> this.props.history.goBack()}>
                            Back
                    </button>
                    
                    {folder && (
                        <h3>
                            {folder.name}
                        </h3>
                    )}
                </div>
                </div>
         )
    }
}

NoteNav.propTypes={
    notes: PropTypes.array,
};
