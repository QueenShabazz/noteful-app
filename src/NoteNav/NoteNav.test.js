import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NoteNav from './NoteNav'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() });

describe(`Form component`, () => {

  it('renders the Form default', () => {
    const wrapper = shallow(
        <NoteNav />
        )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
})