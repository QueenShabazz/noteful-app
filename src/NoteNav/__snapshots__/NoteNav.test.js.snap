// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders the Form default 1`] = `
<div
  className="NoteDetails"
>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
  </div>
</div>
`;
